pkg/distributor/distributor.go

```diff
@@ -510,6 +510,16 @@ func (d *Distributor) Push(ctx context.Context, req *logproto.PushRequest) (*log
 				continue
 			}
 
+			if missing, lbMissing := d.missingEnforcedLabels(stream, tenantID, validationContext); missing {
```

Comment:
I think we should do this after
```
lbs, stream.Labels, stream.Hash, err = d.parseStreamLabels(validationContext, stream.Labels, stream)
```

So we can reuse the `lbs` instead of  implementing `fetchLbs`

Reply:
good catch. pushed a commit addressing it, wdyt?